### PR TITLE
feat(app): add rename to robot settings

### PR DESCRIPTION
**Pull request by koji**
# Overview
Activate Rename function from RobotSettings
close RCORE-578

# Test Plan
design `name robot`
https://www.figma.com/file/AoTLAYuWawlaWItB1umOjr/Release%3A-Opentrons-Flex-Touchscreen-Designs?type=design&node-id=7276-98049&t=nSKrZQV5CKMnB6Sm-0

1. edge config.json
/data/ODD/config.json
unfinishedUnboxingFlowRoute: '/robot-settings/rename-robot'

Settings -> Rename robot shows the above screen (FYI actually this isn't the right path for the ODD app so this is for the only testing purpose)

design `rename robot`
https://www.figma.com/file/AoTLAYuWawlaWItB1umOjr/Release%3A-Opentrons-Flex-Touchscreen-Designs?type=design&node-id=7808-149991&t=nSKrZQV5CKMnB6Sm-0

1. edge config.json
/data/ODD/config.json
unfinishedUnboxingFlowRoute: null

2 Settings -> Rename robot shows the above screen 

# Changelog
- update name robot component
- update SmallButton buttonType since it used `default` and `alt` but now Figma uses `primary` and `secondary` (MediumButton and LargeButton use them too)

# Review requests
Currently using `Link` with `state` to check that the access is from RobotSettings or initial setup since the design is slightly different and back button's path is totally different. However, the functionality is exactly the same.  

# Risk assessment
low

### CODE REVIEW

**Comment by koji** [comment]
- [x] use OnDeviceDisplaySettings `targetPath`
- [x] merge #12610 before going to live

File: app/src/organisms/OnDeviceDisplay/RunningProtocol/ConfirmCancelRunModal.tsx [hunk 1]
```diff
@@ -108,8 +108,8 @@ export function ConfirmCancelRunModal({
           width="100%"
         >
           <SmallButton
-            flex="1"
             buttonType="primary"
+            flex="1"
```
Comment (ewagoner) [inline app/src/organisms/OnDeviceDisplay/RunningProtocol/ConfirmCancelRunModal.tsx]: Why this change?

Comment (koji) [reply app/src/organisms/OnDeviceDisplay/RunningProtocol/ConfirmCancelRunModal.tsx]: oh not necessary

**Comment by ewagoner** [comment]
https://github.com/Opentrons/opentrons/assets/2960/c92558e9-7330-4120-a363-d4d0f51600c0

I was trying to show that the settings page was really slow to update with the new name, but now I can't get the onscreen keyboard to work at all. It worked the first time, but not thereafter.

**Comment by ewagoner** [comment]
> It worked the first time, but not thereafter.

On a reboot, the keyboard started working again and I could change the name multiple times.

**Comment by koji** [comment]
> > It worked the first time, but not thereafter.
> 
> On a reboot, the keyboard started working again and I could change the name multiple times.

Yeah, software keyboard doesn't work well with Remote Dev Tools.
Last week Mel told me that. But at this moment. I haven't dig into the issue since the app on the touchscreen works and the current priority is to add all functionality asap.

File: app/src/pages/OnDeviceDisplay/NameRobot.tsx [hunk 2]
```diff
@@ -153,65 +160,84 @@ export function NameRobot(): JSX.Element {
...
-                lineHeight="1.875rem"
-                fontWeight={TYPOGRAPHY.fontWeightRegular}
-                marginBottom="0.75rem"
-              >
-                {t('name_your_robot_description')}
-              </StyledText>
+              {isInitialSetup ? (
+                <StyledText
+                  color={COLORS.black}
+                  fontSize="1.375rem"
```
Comment (ewagoner) [inline app/src/pages/OnDeviceDisplay/NameRobot.tsx]: use typography constants instead?

File: app/src/pages/OnDeviceDisplay/RobotSettingsDashboard.tsx [hunk 3]
```diff
@@ -258,7 +264,7 @@ const RobotSettingButton = ({
                 color={COLORS.darkGreyEnabled}
                 fontSize="1.375rem"
```
Comment (ewagoner) [inline app/src/pages/OnDeviceDisplay/RobotSettingsDashboard.tsx:265]: same here

**Review by ewagoner — APPROVED** [review]
